libraries/abstractions/secure_sockets/lwip/iot_secure_sockets.c

```diff
@@ -210,11 +210,6 @@ static BaseType_t prvNetworkRecv( void * pvContext,
 
     ctx = ( ss_ctx_t * ) pvContext;
 
-    if( 0 > ctx->ip_socket )
-    {
-        return SOCKETS_SOCKET_ERROR;
-    }
-
     int ret = lwip_recv( ctx->ip_socket,
```

Comment:
Please remind me why we had to remove these? If I recall this was dead/unreachable code right?
I would actually prefer if we can separate out the fixes to the code from the adding of the tests but it may be too late to do that easily? 

Think about the case where we find out this change was bad, if we want to roll it back we will now have to roll back the entire PR.

Also if the PR does not change the code, only tests it is low risk to include it in master, but if it changes code the risk is much higher and this will hold up the review process.

Reply:
this condition always evaluate to true
i created another PR to address it